cdn/signedcookies/sign_cookie_test.go

```diff
@@ -0,0 +1,140 @@
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	key := `nZtRohdNF9m3cKM24IcK4w==`
+	expected := []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+		0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a, 0xe3}
+	if err := ioutil.WriteFile(f.Name(), []byte(key), 0600); err != nil {
+		t.Fatal(err)
+	}
+	b, err := readKeyFile(f.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(b, expected) {
+		t.Fatalf("got=%v; expected=%v", b, expected)
+	}
+}
+
+func TestSignCookie(t *testing.T) {
+	testKey := []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+		0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a, 0xe3} // base64url: nZtRohdNF9m3cKM24IcK4w==
+
+	cases := []struct {
+		testName   string
+		urlPrefix  string
+		keyName    string
+		key        []byte
+		expiration time.Time
+		out        string
+		shouldPass bool
+	}{
+		{
+			// Valid signature
+			testName:   "Valid Domain and Sig",
+			urlPrefix:  "https://media.example.com/segments/",
+			keyName:    "my-key",
+			key:        testKey,
+			expiration: time.Unix(1558131350, 0),
+			out:        "URLPrefix=aHR0cHM6Ly9tZWRpYS5leGFtcGxlLmNvbS9zZWdtZW50cy8=:Expires=1558131350:Keyname=my-key:Signature=QMZgLb8pS9MkhTxcPOQTM5nzJXc=",
+			shouldPass: true,
+		},
+		{
+			// Valid signature, different domain
+			testName:   "Valid Domain #2 & Sig",
+			urlPrefix:  "https://video.example.com/manifests/123/",
+			keyName:    "my-key",
+			key:        testKey,
+			expiration: time.Unix(1558131350, 0),
+			out:        "URLPrefix=aHR0cHM6Ly92aWRlby5leGFtcGxlLmNvbS9tYW5pZmVzdHMvMTIzLw==:Expires=1558131350:Keyname=my-key:Signature=vZdfJ4EnJTsADeKG5-TSwLqdtiw=",
+			shouldPass: true,
+		},
+		{
+			// Mismatched timestamps
+			testName:   "Mismatched Timestamps",
+			urlPrefix:  "https://media.example.com/segments/",
+			keyName:    "my-key",
+			key:        testKey,
+			expiration: time.Unix(1558131350, 0),
+			out:        "URLPrefix=aHR0cHM6Ly9tZWRpYS5leGFtcGxlLmNvbS9zZWdtZW50cy8=:Expires=1858131350:Keyname=my-key:Signature=QMZgLb8pS9MkhTxcPOQTM5nzJXc=",
+			shouldPass: false,
+		},
+		{
+			// Wrong key names
+			testName:   "Mismatched Key Names",
+			urlPrefix:  "https://media.example.com/segments/",
+			keyName:    "bad-key",
+			key:        testKey,
+			expiration: time.Unix(1558131350, 0),
+			out:        "URLPrefix=aHR0cHM6Ly9tZWRpYS5leGFtcGxlLmNvbS9zZWdtZW50cy8=:Expires=1858131350:Keyname=my-key:Signature=QMZgLb8pS9MkhTxcPOQTM5nzJXc=",
+			shouldPass: false,
+		},
+		{
+			// Invalid key material
+			testName:  "Invalid Key Material",
+			urlPrefix: "https://media.example.com/segments/",
+			keyName:   "my-key",
+			key: []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+				0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a},
+			expiration: time.Unix(1558131350, 0),
+			out:        "URLPrefix=aHR0cHM6Ly9tZWRpYS5leGFtcGxlLmNvbS9zZWdtZW50cy8=:Expires=1858131350:Keyname=my-key:Signature=QMZgLb8pS9MkhTxcPOQTM5nzJXc=",
+			shouldPass: false,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(
```

Comment:
One line. Could declare the test name separately if it is too long. The indentation is a little hard for me to follow with this extra line break.